### PR TITLE
refactor: use `mark_safe` more sparsely, replace `format` with `format_html`

### PR DESCRIPTION
**Pull request by WofWca**
## Proposed changes

This should make it easier to maintain the code without introducing security vulnerabilities.

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

File: weblate/trans/models/label.py [hunk 1]
```diff
@@ -46,10 +46,8 @@ class Meta:
         verbose_name_plural = "label"
 
     def __str__(self):
-        return mark_safe(
-            '<span class="label label-{}">{}</span>'.format(
-                self.color, escape(self.name)
-            )
+        return format_html(
+            '<span class="label label-{}">{}</span>', mark_safe(self.color), self.name
```
Comment (nijel) [inline weblate/trans/models/label.py:50]:
```suggestion
            '<span class="label label-{}">{}</span>', self.color, self.name
```

I'd omit `mark_safe` here even though the color list is currently hard-coded (see https://github.com/WeblateOrg/weblate/blob/main/weblate/utils/colors.py).

File: weblate/trans/views/dashboard.py [hunk 2]
```diff
@@ -177,11 +177,10 @@ def home(request):
...
         messages.warning(
             request,
-            mark_safe(
-                '<a href="{}">{}</a>'.format(
-                    reverse("profile") + "#account",
-                    escape(_("Please set your full name and e-mail in your profile.")),
-                )
+            format_html(
+                '<a href="{}">{}</a>',
+                mark_safe(reverse("profile") + "#account"),
```
Comment (nijel) [inline weblate/trans/views/dashboard.py:182]:
```suggestion
                reverse("profile") + "#account",
```

**Comment by WofWca** [comment]
I'll probably do more of this (in another MR), if nobody minds.

File: weblate/trans/context_processors.py [hunk 3]
```diff
@@ -169,11 +169,10 @@ def weblate_context(request):
...
         "weblate_name_link": mark_safe(f'<a href="{escape(WEBLATE_URL)}">Weblate</a>'),
-        "weblate_version_link": mark_safe(
-            '<a href="{}">Weblate {}</a>'.format(
-                escape(WEBLATE_URL),
-                "" if settings.HIDE_VERSION else weblate.utils.version.VERSION,
-            )
+        "weblate_version_link": format_html(
+            '<a href="{}">Weblate {}</a>',
+            WEBLATE_URL,
+            mark_safe("" if settings.HIDE_VERSION else weblate.utils.version.VERSION),
```
Comment (nijel) [inline weblate/trans/context_processors.py:175]:
```suggestion
            "" if settings.HIDE_VERSION else weblate.utils.version.VERSION,
```

**Review by nijel — COMMENTED** [review]
Looks good, I was not aware of `format_html`...

I'd remove mark_safe usage on strings which are currently assumed to be safe - the code is easier to read without it and the performance penalty should not be noticeable.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**